pkg/scheduler/objects/preemption.go

```diff
@@ -155,7 +155,10 @@ func (p *Preemptor) initWorkingState() {
 			}
 			allocationsByNode[nodeID] = append(allocations, allocation)
 			queueByAlloc[allocation.GetAllocationKey()] = victims
-			nodeInstanceTypeMap[nodeID] = allocation.GetInstanceType()
+			instType := allocation.GetInstanceType()
+			if instType != UnknownInstanceType {
```

Comment:
Just to code defensively, I would skip the update if the allocation type is either unknown *or* empty. This should ensure we don't clobber the instance type. I know we're confident all the allocation instance types have been set, but there may be edge cases (recovery for instance).

Reply:
Thanks a lot @craigcondit I'm working on a new rev.

Reply:
Hi @craigcondit new rev uploaded.  Many thanks again!